wit/couchbase/fts.wit

```diff
@@ -0,0 +1,203 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Support Full Text Search (FTS) provided by Couchbase, which spans:
+///   - Vectors
+///   - Structured/Unstructured text
+///   - Dates
+///   - Numbers
+///   - CIDR notation
+///   - Geospatial data
+///
+/// Reference: https://docs.couchbase.com/server/current/search/search.html
+interface fts {
+  use types.{
+    request-span, document, document-field-name, collection, sort-direction, 
+    bucket-error, search-index-name, retry-strategy, mutation-state
+    };
+
+  variant fts-bucket-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Buckets for use with FTS
+  ///
+  /// NOTE: This is unnecessary (and may be implemented as a no-op) for Couchbase versions greater than 6.5
+  resource fts-bucket {
+    constructor(name: string);
+    /// Wait until a bucket is ready to use
+    ///
+    /// This should be done on couchbase versions earlier than 6.5, before performing any search queries.
+    wait-until-ready: func() -> result<_, bucket-error>;
+  }
+
+  /// Errors that occur when creating an FTS index
+  variant fts-index-create-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Errors that occur when creating an FTS index
+  variant fts-bucket-create-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Errors that occur when performing an FTS search
+  variant fts-search-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Search facet
+  variant facet {
+    /// Term Facet
+    term(tuple<string, u64>),
+    /// Numeric Facet
+    numeric(tuple<string, u64>),
+    /// Date range
+    date-range(tuple<string, string, string>),
+  }
+
+  /// Options to use when performing a search
+  record search-options {
+    /// Number of results to limit the search results by
+    limit: u32,
+
+    /// Level of data consistency desired for a search query
+    scan-consistency: u64,
```

Comment:
Why not use an enum here ? Like you did for queries with `query-scan-consistency` ?

Reply:
We definitely could -- will update and use an enum!

- [ ] Use an enum for `scan-consistency` under `fts`